src/subcommand/init.rs

```diff
@@ -212,6 +241,17 @@ function _zoxide_hook --on-event fish_prompt
 end
 "#;
 
+const POSIX_HOOK_PROMPT: &str = r#"
+_zoxide_hook() {
+    zoxide add
```

Comment:
Given that anything this hook writes to `stdout` becomes part of the user's prompt (including debug statements, if someone is ever working on `zoxide` with a POSIX shell), I think it would be wise to redirect `stdout` elsewhere (either to `/dev/null` or `stderr`).

Reply:
I'll proceed under the assumption that we don't want to redirect `stderr` to `/dev/null`.